src/test/scala/edu/ie3/simona/model/participant/evcs/EvcsModelSpec.scala

```diff
@@ -0,0 +1,973 @@
+        )
+
+        forAll(cases) {
+          (
+              lastStored1,
+              lastStored2,
+              lastPower2,
+              expectedPRef,
+              expectedPMin,
+              expectedPMax
+          ) =>
+            // stays one more hour
+            val ev1 = EvModelWrapper(
+              new MockEvModel(
+                UUID.randomUUID(),
+                "Mock EV 1",
+                10.0.asKiloWatt, // AC is relevant,
+                20.0.asKiloWatt, // DC is not
+                10.0.asKiloWattHour,
+                lastStored1.asKiloWattHour,
+                10800L
+              )
+            )
+
+            // has not been charging before
+            val schedule1 = SortedSet(
+              ScheduleEntry(3600L, 7200L, Kilowatts(0d))
+            )
+
+            // stays two more hours
+            val ev2 = EvModelWrapper(
+              new MockEvModel(
+                UUID.randomUUID(),
+                "Mock EV 2",
+                5.0.asKiloWatt, // AC is relevant,
+                10.0.asKiloWatt, // DC is not
+                15.0.asKiloWattHour,
+                lastStored2.asKiloWattHour,
+                14400L
+              )
+            )
+
+            // has been charging for 1.5 hours with given power
+            val schedule2 = SortedSet(
+              ScheduleEntry(0L, 5400L, Kilowatts(lastPower2))
+            )
+
+            evcsModel.determineFlexOptions(
+              data,
+              EvcsState(
+                Seq(ev1, ev2),
+                Map(ev1.uuid -> schedule1, ev2.uuid -> schedule2),
+                0L
+              )
+            ) match {
+              case ProvideMinMaxFlexOptions(
+                    modelUuid,
+                    refPower,
+                    minPower,
+                    maxPower
+                  ) =>
+                modelUuid shouldBe evcsModel.getUuid
+                refPower should approximate(Kilowatts(expectedPRef))
+                minPower should approximate(Kilowatts(expectedPMin))
+                maxPower should approximate(Kilowatts(expectedPMax))
+            }
+        }
+
+      }
+
+      "charging with maximum power and allowing v2g" in {
+        val evcsModel =
+          evcsStandardModel.copy(strategy = ChargingStrategy.MAX_POWER)
+
+        val currentTick = 7200L
+
+        val data = EvcsRelevantData(
+          currentTick,
+          Seq.empty
+        )
+
+        val cases = Table(
+          (
+            "lastStored1",
+            "lastStored2",
+            "lastPower2",
+            "expectedPRef",
+            "expectedPMin",
+            "expectedPMax"
+          ),
+
+          /* 1: empty */
+          // 2: empty
+          (0.0, 0.0, 0.0, 15.0, 15.0, 15.0),
+          // 2: at lower margin
+          (0.0, 3.0, 0.0, 15.0, 10.0, 15.0),
+          // 2: mid-way full (charged to 7.5 kWh), forced charging
+          (0.0, 0.0, 5.0, 15.0, 10.0, 15.0),
```

Comment:
lastStored2 = 7.5? 
```suggestion
          // 2: mid-way full (charged to 7.5 kWh), forced charging
          (0.0, 7.5, 5.0, 15.0, 10.0, 15.0),
```

Reply:
Nope, because we charge from 0 kWh to 7.5 kWh (as the comment says) because we charge with 5 kWh (next parameter) for 1,5 hours (from tick 0 to tick 5400, line 605).
Charing happens _before_ calculating flex options here because it's part of calculating the current state.